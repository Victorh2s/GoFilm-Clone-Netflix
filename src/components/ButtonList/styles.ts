import styled, { css } from 'styled-components';

export const button2 = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 10%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};


  :hover{
    background: rgba(255,255,255, 20%);
  }

  `}
`;

export const link2 = styled.a`
  ${({ theme }) => css`
  display: flex;
  text-decoration: none;
  justify-content: center;
  align-items: center;

>svg{
  width: 4rem;
  color: ${theme.colors.white};
  padding: 0 0.5rem;
}


  `}
`;
