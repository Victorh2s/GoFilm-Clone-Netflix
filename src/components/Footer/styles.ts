import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    background: ${theme.colors.dark};
    height: 10rem;

  `}
`;
export const Container = styled.section`
  ${({ theme }) => css`

    display: flex;
    text-align: center;
    justify-content: center;
    padding: ${theme.spacings.small};


  `}
`;

export const link = styled.a`
  ${({ theme }) => css`

  padding:${theme.spacings.small};

  >svg{
    width: 3rem;
    height: 4rem;
    color: ${theme.colors.white};
  }

  `}
`;
