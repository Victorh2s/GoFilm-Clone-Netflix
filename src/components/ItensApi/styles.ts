import styled, { css } from 'styled-components';

export const itens = styled.div`
  ${({ theme }) => css`

  font-size: 1.8rem;
  font-weight: bold;
  margin-top: 1.5rem;
  color: ${theme.colors.white};

  `}
`;

export const Vote = styled.div`
  ${({ theme }) => css`
  display: inline-block;
  color: ${theme.colors.green};
  margin-right: 1.5rem;


  `}
`;

export const Year = styled.div`
  display: inline-block;
  margin-right: 1.5rem;
`;

export const Season = styled.div`
  display: inline-block;
`;

export const Description = styled.div`
  ${({ theme }) => css`

  margin-top: 1.5rem;
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.mediumGray};
  max-width: 35%;


  `}
`;

export const Genres = styled.div`
  ${({ theme }) => css`

  margin-top: 1.3rem;
  color: ${theme.colors.mediumGray};
  font-size: ${theme.font.sizes.small};
  font-style: italic;

  `}
`;
