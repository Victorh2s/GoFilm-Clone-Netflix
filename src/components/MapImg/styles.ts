import styled from 'styled-components';

export const MovieFilm = styled.div`
  margin-left: 1rem;

`;

export const Link = styled.a`

`;

export const Image = styled.img`
  width: 18rem;
  transform: scale(0.9);
  transition: all 200ms ease-in-out;
  cursor: pointer;
  :hover{
    transform: scale(1);
  }


`;
