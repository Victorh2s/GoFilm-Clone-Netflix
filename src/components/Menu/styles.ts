import styled, { css } from 'styled-components';

export const Menu = styled.menu`
  ${({ theme }) => css`

  display: flex;
  height: 6rem;
  top: 0;
  left:0;
  right: 0;
  background: transparent;
  align-items: center;
  justify-content: space-between;
  overflow: hidden;
  position: fixed;
  transition: all 200ms ease-in-out;
  z-index: 4;



    `}
`;

export const Logo = styled.a`
  padding: 5rem;
`;

export const Netflix = styled.img`
  ${({ theme }) => css`
  width: 12rem;
  height: 12rem;
  `}
`;

export const Avatar = styled.a`
  padding: 5rem;
`;

export const icon = styled.img`
  ${({ theme }) => css`
  width: 3rem;
  height: 3rem;
  `}
`;
