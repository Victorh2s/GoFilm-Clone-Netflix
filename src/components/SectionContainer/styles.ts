import styled, { css } from 'styled-components';

export const SectionContainer = styled.div`
  ${({ theme }) => css`

  max-width: 120rem;
  margin: 0 auto;
  padding: ${theme.spacings.large};

  `}
`;
