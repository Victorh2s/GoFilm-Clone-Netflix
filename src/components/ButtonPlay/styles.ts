import { Wrapper as Text } from '../Text/styles';
import styled, { css } from 'styled-components';

export const button = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 70%);
  padding: 0.5rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};

  :hover{
    background: rgba(255,255,255, 80%);
  }

  `}
`;

export const link = styled.a`
  ${({ theme }) => css`
  text-decoration: none;
  justify-content: center;
  align-items: center;
  display: flex;

  ${Text}{
    color: ${theme.colors.dark};
  }

>svg{
  width: 4rem;
  color: black;
  padding: 0 0.5rem;
}
text-decoration: none;
color: ${theme.colors.dark};

  `}
`;
