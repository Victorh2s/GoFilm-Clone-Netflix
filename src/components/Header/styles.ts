import { Wrapper as Text } from '../Text/styles';
import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Wrapper = styled.section`
    height: 97vh;
`;

export const GladientVertical = styled.div`
width: inherit;
height:inherit;
background: linear-gradient(to top, #0d0c0c 5%, transparent 100%);

`;

export const GladientHorizontal = styled.div`

width: inherit;
height:inherit;
background: linear-gradient(to right , #0d0c0c 18%, transparent 95%);
display: flex;
flex-direction: column;
justify-content: center;
padding-left: 3rem;
padding-top: 4rem;

${Heading}{
  max-width: 50%;
}

`;

export const itens = styled.div`
  ${({ theme }) => css`

  font-size: 1.8rem;
  font-weight: bold;
  margin-top: 1.5rem;
  color: ${theme.colors.white};

  `}
`;

export const Vote = styled.div`
  ${({ theme }) => css`
  display: inline-block;
  color: ${theme.colors.green};
  margin-right: 1.5rem;


  `}
`;

export const Year = styled.div`
  display: inline-block;
  margin-right: 1.5rem;

`;

export const Season = styled.div`
  display: inline-block;

`;

export const Description = styled.div`
  ${({ theme }) => css`

  margin-top: 1.5rem;
  font-size: ${theme.font.sizes.small};
  color: ${theme.colors.mediumGray};
  max-width: 35%;


  `}
`;

export const Genres = styled.div`
  ${({ theme }) => css`

  margin-top: 1.3rem;
  color: ${theme.colors.mediumGray};
  font-size: ${theme.font.sizes.small};
  font-style: italic;

  `}
`;

export const buttons = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 70%);
  padding: 1rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};

  :hover{
    background: rgba(255,255,255, 80%);
  }

  `}
`;

export const link = styled.a`
  ${({ theme }) => css`
  text-decoration: none;
  justify-content: center;
  align-items: center;
  display: flex;

  ${Text}{
    color: ${theme.colors.dark};
  }

>svg{
  width: 4rem;
  color: black;
  padding: 0 0.5rem;
}
text-decoration: none;
color: ${theme.colors.dark};

  `}
`;

export const buttons2 = styled.button`
  ${({ theme }) => css`

  border: none;
  background: rgba(255,255,255, 10%);
  padding: 1rem 2rem;
  margin-top: 3rem;
  cursor: pointer;
  transition: all 200ms ease-in-out;
  font-weight: bold;
  font-size:  ${theme.font.sizes.small};


  :hover{
    background: rgba(255,255,255, 20%);
  }

  `}
`;

export const link2 = styled.a`
  ${({ theme }) => css`
  display: flex;
  text-decoration: none;
  justify-content: center;
  align-items: center;

>svg{
  width: 4rem;
  color: ${theme.colors.white};
  padding: 0 0.5rem;
}


  `}
`;
