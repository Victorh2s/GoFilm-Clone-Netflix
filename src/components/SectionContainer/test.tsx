import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { SectionContainer, SectionContainerProps } from '.';

const props: SectionContainerProps = {
  children:
    'Lorem ipsum dolor sit, amet consectetur adipisicing elit. A illo possimus minus aliquid nemo minima, doloribus harum eius sint recusandae alias eum rerum ab neque et, blanditiis, molestiae exercitationem quas?',
};

describe('<SectionContainer />', () => {
  it('should render', () => {
    renderTheme(<SectionContainer {...props} />);

    expect(screen.getByText(/Lorem ipsum/i)).toBeInTheDocument();
  });

  it('should match to snapshot', () => {
    const { container } = renderTheme(<SectionContainer {...props} />);

    expect(container).toMatchSnapshot();
  });
});
