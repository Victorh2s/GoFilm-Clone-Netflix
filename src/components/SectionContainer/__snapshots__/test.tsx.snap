// Vitest Snapshot v1

exports[`<SectionContainer /> > should match to snapshot 1`] = `
.c0 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

<div>
  <div
    class="c0"
  >
    Lorem ipsum dolor sit, amet consectetur adipisicing elit. A illo possimus minus aliquid nemo minima, doloribus harum eius sint recusandae alias eum rerum ab neque et, blanditiis, molestiae exercitationem quas?
  </div>
</div>
`;
